### PR TITLE
Adds range plugin for infrared and ultrasound sensors from PAL Robotics

### PR DESCRIPTION
**Pull request by bmagyar**
Rebase of #311 to indigo-devel + added install rule for new plugin and squashed 'em together.

Solves #162 on ROS Indigo.

### CODE REVIEW

**Comment by bmagyar** [comment]
@j-rivero

File: gazebo_plugins/src/gazebo_ros_range.cpp [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+    ROS_INFO("Laser plugin missing <frameName>, defaults to /world");
+    this->frame_name_ = "/world";
+  }
+  else
+    this->frame_name_ = this->sdf->Get<std::string>("frameName");
+
+  if (!this->sdf->HasElement("topicName"))
+  {
+    ROS_INFO("Laser plugin missing <topicName>, defaults to /world");
+    this->topic_name_ = "/world";
```
Comment (gavanderhoorn) [inline gazebo_plugins/src/gazebo_ros_range.cpp]: Is this copy/pasta from the `frameName` check above? I'm really not a Gazebo plugin expert, but this seems like a strange topic name for a sonar sensor?

Comment (bmagyar) [inline gazebo_plugins/src/gazebo_ros_range.cpp]: Oh it is indeed copy pasta with tomato sauce. Also the message there make no sense, I shall fix them shortly.

**Comment by bmagyar** [comment]
@scpeters, what do you think?

**Comment by scpeters** [comment]
+1

**Comment by rethink-imcmahon** [comment]
+1 great addition @AlexReimann & @bmagyar

**Comment by j-rivero** [comment]
Amazing! Thanks Ben! Would you mind to include a quick test world file in `gazebo_plugins/test/range`?

**Comment by bmagyar** [comment]
Only a .world file? If I need anything else, please you point me at an example from the test folder.

**Comment by j-rivero** [comment]
The point is to have something to demonstrate and run the code for testing and learning proposes. I think that a world file + launch file could be a good combo. Give a look to what Nate implemented in pull request for #338.

Automatic testing would be ideal, but we probably need to setup a proper mechanism in place first to be shared by the different plugins.

**Comment by bmagyar** [comment]
Took some time to get back here, but I finally managed to add a world file + a launch file.

There are 2 boxes in the world, both with a range sensor. One looks at the other, the other looks into the horizon like a true western hero.

To launch and check:

`roslaunch gazebo_ros range.world.launch`
`rostopic echo /sonar`
`rostopic echo /sonar2`

There is a gaussian noise on the sensors, so the perfect 0.25 measurement hardly appears, but it works well.

Now I will try to add a rostest in a new commit.

**Comment by j-rivero** [comment]
> Took some time to get back here, but I finally managed to add a world file + a launch file.

Thanks very much Bence for taking the time to implement it. I was able to build your patch built on top of pull request #350 using gazebo5 and gazebo6 on Trusty/Indigo. 

> Now I will try to add a rostest in a new commit.

I will pay the beers if you made that :) Since I'm planning on releasing (issue #335), I want to get this in, so I will merge it just after pull request #350. Feel free to send a new pull request for the rostest support.

**Comment by bmagyar** [comment]
Great!
This means that gazebo plugins will be synced for now between indigo and
jade? Good news for my patch for the block laser ;)

**Comment by j-rivero** [comment]
> This means that gazebo plugins will be synced for now between indigo and jade?

That should have been done in pull request #350 and this PR will go into jade in pull request #352

**Pull request merged**